src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.Arrays;
```

Comment:
The use of `Arrays.copyOf()` in the `grow` method violates the constraint of not using utility classes. You should implement array copying manually.